src/App.tsx

```diff
@@ -65,12 +68,9 @@ const App: React.FC = () => {
 							element={
 								isLoggedIn ? (
 									hasCourses ? (
-										<Courses
-											courses={coursesList}
-											authors={mockedAuthorsList}
-										/>
+										<Courses />
 									) : (
-										<EmptyCoursesList />
+										<EmptyCoursesList userRole={userRoleFromStore} />
 									)
 								) : (
 									<Navigate to='/login' />
```

Comment:
Also if you want to redirect based on some value you can take into consideration protected routes 

https://dev.to/collins87mbathi/reactjs-protected-route-m3j#:~:text=To%20create%20a%20protected%20route,redirected%20to%20the%20login%20page.